src/modes/hip/device.cpp

```diff
@@ -225,12 +225,14 @@ namespace occa {
     }
 
     void device::setArchCompilerFlags(occa::properties &kernelProps) {
-      const std::string compiler_flags = kernelProps.get<std::string>("compiler_flags");
+      const std::string hipccCompilerFlags = (
+        kernelProps.get<std::string>("hipcc_compiler_flags")
+      );
 
-      if (compiler_flags.find("-t gfx") == std::string::npos
-          && compiler_flags.find("-arch=sm") == std::string::npos) {
-        kernelProps["compiler_flags"] += ' ';
-        kernelProps["compiler_flags"] += kernelProps["compiler_flag_arch"];
+      if (hipccCompilerFlags.find("-t gfx") == std::string::npos
+          && hipccCompilerFlags.find("-arch=sm") == std::string::npos) {
+        kernelProps["hipcc_compiler_flags"] += ' ';
```

Comment:
I think this should be " " not ' '. The json complains about adding a single char type.

Reply:
Good catch!